Lib9c/PolicyAction/Tx/Begin/AutoJoinGuild.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Nekoyume.Action;
+using Nekoyume.Action.Guild.Migration.Controls;
+using Nekoyume.Extensions;
+using Nekoyume.Model.State;
+using Nekoyume.Module;
+using Nekoyume.Module.Guild;
+using Nekoyume.TypedAddress;
+using Serilog;
+
+namespace Nekoyume.PolicyAction.Tx.Begin
+{
+    /// <summary>
+    /// An action that automatically joins to Planetarium guild if it contracted pledge with Planetarium.
+    /// </summary>
+    public class AutoJoinGuild : ActionBase
+    {
+        public override IValue PlainValue => Null.Value;
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            throw new InvalidOperationException("Policy action shouldn't be serialized.");
+        }
+
+        public override IWorld Execute(IActionContext context)
+        {
+            if (!context.IsPolicyAction)
+            {
+                throw new InvalidOperationException(
+                    "This action must be called when it is a policy action.");
+            }
+
+            var world = context.PreviousState;
+            var signer = context.GetAgentAddress();
+
+            var (resultWorld, _) = GuildMigrationCtrl.MigratePlanetariumPledgeToGuild(world, signer);
+            return resultWorld;
```

Comment:
In general (TryParse, TryGetValue, etc.), if the return type is bool, there should be a guarantee that this method will not throw any exceptions.
I think it would be a bit confusing for someone using this method to have a mix of successes, failures, and exceptions.
What I would suggest is to define and throw an exception class that can be used in the case of returning failure (true).
Then, in AutoJoinGuild, make sure that only that exception is handled by the catch.